WalletWasabi/Tor/TorMonitor.cs

```diff
@@ -49,6 +64,12 @@ public TorMonitor(TimeSpan period, Uri fallbackBackendUri, TorProcessManager tor
 
 	private Task? BootstrapTask { get; set; }
 
+	/// <summary>Whether we should try Tor process restart to fix <see cref="ReplyType.TtlExpired"/> issues.</summary>
+	private bool TryTorRestart { get; set; }
```

Comment:
I think you should initialize this as `true` instead of `false`. This should also explain why @yahiheb's test is negative.